src/index.js

```diff
@@ -38,29 +39,35 @@ function buildValidations(referenced, dereferenced, receivedOptions) {
     const options = getOptions(receivedOptions);
 
     const schemas = {};
-    Object.keys(dereferenced.paths).forEach(function (currentPath) {
-        const parsedPath = dereferenced.basePath && dereferenced.basePath !== '/'
-            ? dereferenced.basePath.concat(currentPath.replace(/{/g, ':').replace(/}/g, ''))
-            : currentPath.replace(/{/g, ':').replace(/}/g, '');
-        schemas[parsedPath] = {};
-        Object.keys(dereferenced.paths[currentPath])
-            .filter(function (parameter) { return parameter !== 'parameters' })
-            .forEach(function (currentMethod) {
-                const parsedMethod = currentMethod.toLowerCase();
-
-                let requestValidator;
-                if (options.buildRequests) {
-                    requestValidator = buildRequestValidator(referenced, dereferenced, currentPath,
-                        parsedPath, currentMethod, options);
-                }
 
-                let responseValidator;
-                if (options.buildResponses) {
-                    responseValidator = buildResponseValidator(referenced, dereferenced, currentPath, parsedPath, currentMethod, options);
-                }
+    const basePaths = dereferenced.servers
+        ? dereferenced.servers.map(({ url }) => URL.parse(url).pathname)
+        : [dereferenced.basePath || '/'];
 
-                schemas[parsedPath][parsedMethod] = Object.assign({}, requestValidator, { responses: responseValidator });
-            });
+    Object.keys(dereferenced.paths).forEach(function (currentPath) {
+        basePaths.forEach(basePath => {
+            const fullPath = basePath.replace(/\/$/, '').concat(currentPath);
```

Comment:
Maybe this logic could be extracted into an internal function for easier reading?